Sources/PrivacyStats/PrivacyStats.swift

```diff
@@ -0,0 +1,268 @@
+//
+//  PrivacyStats.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Combine
+import Common
+import Foundation
+import os.log
+import Persistence
+import TrackerRadarKit
+#if os(macOS)
+import AppKit
+#elseif os(iOS)
+import UIKit
+#endif
+
+/**
+ * Errors that may be reported by `PrivacyStats`.
+ */
+public enum PrivacyStatsError: CustomNSError {
+    case failedToFetchPrivacyStatsSummary(Error)
+    case failedToStorePrivacyStats(Error)
+    case failedToLoadCurrentPrivacyStats(Error)
+
+    public static let errorDomain: String = "PrivacyStatsError"
+
+    public var errorCode: Int {
+        switch self {
+        case .failedToFetchPrivacyStatsSummary:
+            return 1
+        case .failedToStorePrivacyStats:
+            return 2
+        case .failedToLoadCurrentPrivacyStats:
+            return 3
+        }
+    }
+
+    public var underlyingError: Error {
+        switch self {
+        case .failedToFetchPrivacyStatsSummary(let error),
+                .failedToStorePrivacyStats(let error),
+                .failedToLoadCurrentPrivacyStats(let error):
+            return error
+        }
+    }
+}
+
+/**
+ * This protocol describes database provider consumed by `PrivacyStats`.
+ */
+public protocol PrivacyStatsDatabaseProviding {
+    func initializeDatabase() -> CoreDataDatabase
+}
+
+/**
+ * This protocol describes `PrivacyStats` interface.
+ */
+public protocol PrivacyStatsCollecting {
+
+    /**
+     * Record a tracker for a given `companyName`.
+     *
+     * `PrivacyStats` implementation calls the actor under the hood,
```

Comment:
To ease understanding think we could say here:

```suggestion
     * `PrivacyStats` implementation calls the `CurrentPack` actor under the hood,
```